csharp/src/Apache.Arrow/C/CArrowArrayStream.cs

```diff
@@ -35,15 +35,23 @@ public unsafe struct CArrowArrayStream
         ///
         /// Return value: 0 if successful, an `errno`-compatible error code otherwise.
         ///</summary>
-        public delegate* unmanaged[Stdcall]<CArrowArrayStream*, CArrowSchema*, int> get_schema;
```

Comment:
If we make these function pointers `internal`, how are users of these APIs supposed to call them?

Reply:
They should not; the importer will take care of calling them.

Reply:
I don't fully follow. If I wanted to take some C arrow data (let's say coming from C++) and natively work with it - not using the "normal" C# ArrowTypes, Schema, RecordBatches, etc. but instead manually calling into the native functions on the CArrowArrayStream, that is no longer possible? Why not?

The reason for interoping at the native layer would be for performance - I wouldn't need to allocate a bunch of managed objects just to interact with the Arrow information coming from some other library.

Reply:
See https://github.com/apache/arrow/pull/35810#issuecomment-1584531095 for an explanation on why the function pointer fields are internal.

Reply:
Another option would be to only publicly expose the function pointer fields on `net5.0+`. Keep them `internal` for netstandard and netfx where they need to be declared `Cdecl` (and honestly function pointers aren't really meant to be used on netfx and netstandard since they came in C# 9 - https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/configure-language-version).

That way users on net5+ can still call the function pointers, if they need to. For netstandard and netfx, they are no worse off - they can't call them with this change anyway.